### PR TITLE
feat: Update the metrics crate to 0.15

### PR DESCRIPTION
**Pull request by Marwes**
BREAKING CHANGE

Users should update their metrics version to 0.15 or metrics will not be
reported

### CODE REVIEW

**Comment by bergmark** [comment]
What about @ramn's suggestion

> Could let metrics_cloudwatch builder take a function pointer to https://docs.rs/metrics/0.14.2/metrics/fn.set_boxed_recorder.html and use that to initialize. That should give us a compilation error when metrics_cloudwatch tries to pass in a Recorder from the wrong crate.

These updates will still be breaking changes then, but at least they will be API breaking instead of runtime breaking.

**Review by bergmark — APPROVED** [review]
Do you usually do version/changelog updates separately?

**Comment by ramn** [comment]
> Do you usually do version/changelog updates separately?

I think Markus uses https://github.com/sunng87/cargo-release

**Review by ramn — APPROVED** [review]
LGTM

Any concerns @bergmark ?

**Comment by bergmark** [comment]
No, i approved it

**Review by ramn — APPROVED** [review]

File: src/collector.rs [hunk 1]
```diff
@@ -154,7 +154,7 @@ pub(crate) async fn init_future(
     Ok(())
 }
 
-pub(crate) fn new(config: Config) -> (RecorderHandle, impl Future<Output = ()>) {
```
Comment (ramn) [inline src/collector.rs]: Why was this necessary?

Comment (Marwes) [reply src/collector.rs]: It was used in benchmarks

Comment (ramn) [reply src/collector.rs]: I don't like that we expose it in the public API :/
